sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -314,3 +315,24 @@ def test_backward_neg_tol():
 
     assert 0 < sfs.get_support().sum() < X.shape[1]
     assert new_score < initial_score
+
+
+def test_cv_generator_support():
+    """Check that no exception raised when cv is generator
+
+    non-regression test for #25957
+    """
+    X, y = make_classification()
+
+    groups = np.zeros_like(y, dtype=int)
+    groups[y.size // 2 :] = 1
+
+    cv = LeaveOneGroupOut()
+    splits = cv.split(X, y, groups=groups)
+
+    knc = KNeighborsClassifier(n_neighbors=5)
+
+    sfs = SequentialFeatureSelector(
+        knc, n_features_to_select=5, scoring="accuracy", cv=splits
+    )
```

Comment:
Nit: To simplify the test because accuracy is already the default metric:

```suggestion
    sfs = SequentialFeatureSelector(knc, n_features_to_select=5, cv=splits)
```